lib/mix/test/mix/cli_test.exs

```diff
@@ -60,6 +60,13 @@ defmodule Mix.CLITest do
     end
   end
 
+  test "tasks with slashes in them raise a NoTaskError right away" do
+    in_fixture "no_mixfile", fn ->
+      contents = mix ~w[my/task]
+      assert contents =~ "** (Mix) The task my/task could not be found"
```

Comment:
Please rebase because on master the error message is `The task "my/task" ...` :)